src/zcbenchmarks.cpp

```diff
@@ -417,16 +417,33 @@ double benchmark_increment_note_witnesses(size_t nTxs)
 }
 
 // Fake the input of a given block
-class FakeCoinsViewDB : public CCoinsViewDB {
+// This class is based on the class CCoinsViewDB, but with limited functionality.
+// The construtor and the functions `GetCoins` and `HaveCoins` come directly from
+// CCoinsViewDB, but the rest are either mocks and/or don't really do anything.
+class FakeCoinsViewDB : public CCoinsView {
```

Comment:
What is the purpose of this change? I don't see how it relates to the PR.

Reply:
This could potentially be done as part of another PR, but I felt as though it fell under the umbrella of this one (I mention something in the issue that this resolves). The benchmark, `benchmark_connectblock_slow`, which relied on this had may have unintended behavior based on the changes made in `CCoinsView` with the sapling upgrade (abstract methods were added to `CCoinsView`, and overwritten in `CCoinsViewDB`. `FakeCoinsViewDB` would have defaulted to whatever `CCoinsViewDB` did when those methods were called which I think was counter to the point of the class). I changed the inheritance hierarchy to try to make this impossible in the future.

I didn't make further changes to `benchmark_connectblock_slow` because it depends on a file that is independent of the repo. For that reason I was unable to test whether or not this benchmark worked either before or after the change. I think the changes I made are a step in the right direction, but all that being said it might make sense to create a separate issue to look in to this benchmark.